frontend/packages/ux-editor/src/components/Properties/PropertiesHeader/EditLayoutSetForSubform/EditLayoutSet/CreateNewSubformSection/SubformDataModel.tsx

```diff
@@ -2,30 +2,32 @@ import React from 'react';
 import { StudioTextfield, StudioNativeSelect, StudioProperty } from '@studio/components';
 import { LinkIcon } from '@studio/icons';
 import { useTranslation } from 'react-i18next';
-import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
-import { useAppMetadataModelIdsQuery } from 'app-shared/hooks/queries/useAppMetadataModelIdsQuery';
 import classes from './SubformDataModel.module.css';
 
 export type SubformDataModelProps = {
   setDisplayDataModelInput: (setDisplayDataModelInput: boolean) => void;
-  setNewDataModel: (dataModelId: string) => void;
   displayDataModelInput: boolean;
   setSelectedDataModel: (dataModelId: string) => void;
+  dataModelIds?: string[] | undefined;
+  validateName?: (name: string) => void;
+  dataModelNameError: string | undefined;
+  setIsTextfieldEmpty?: (isEmpty: boolean) => void;
```

Comment:
```suggestion
  dataModelIds?: string[];
  validateName: (name: string) => void;
  dataModelNameError: string;
  setIsTextfieldEmpty: (isEmpty: boolean) => void;
```
I think there can be done some adjustments of the types here.

- `dataModelIds?` already implies undefined, so `| undefined` is unnecessary.
- As far I can see `validateName`, `dataModelNameError`, and `setIsTextfieldEmpty` are always provided in `SubformDataModel` and don't need to be optional.